src/D2L.Security.OAuth2/Keys/Default/MonoRsaSecurityKey.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace D2L.Security.OAuth2.Keys.Default {
+
+	internal sealed class MonoRsaSecurityKey : AsymmetricSecurityKey {

```

Comment:
No, there are type checks to see if you are using one or the other. I could do the type check for the mono only version, since that's the only one you should be using.